scispacy/custom_sentence_segmenter.py

```diff
@@ -1,122 +1,51 @@
-            last_open_paren = close_paren_to_stack[token.text].pop()
-            if prev_tokens[-1] and prev_tokens[-1].text != ".":
-                # allow things like (A) or [A] to start a sentence
-                if not (last_open_paren.i == (token.i-2) and len(prev_tokens[-1].text) == 1):
-                    doc[last_open_paren.i].is_sent_start = False
-
-        # can't start a sentence on a matched "
-        # for example: She turned to him, "This is great." She held the book out to show him.
-        # The second quote cannot start a sentence
-        if token.text == '"':
-            if double_quotes_stack != []:
-                double_quotes_stack.pop()
-                doc[token.i].is_sent_start = False
-            else:
-                double_quotes_stack.append(token)
-
-        # handling the case of a capital letter after a ) unless that was preceeded by a .
-        # for example: 'the support of the Defense Advanced Resarch Projects Agency (DARPA) Deep Exploration'
-        first_char = token.text[0]
-        if first_char.isupper():
-            if prev_tokens[-1] and prev_tokens[-1].text == ')':
-                if prev_tokens[-2] and prev_tokens[-2].text != ".":
-                    doc[token.i].is_sent_start = False
-
-        # can't start a new sentence immediately after a token ending in a digit or character
-        if prev_tokens[-1] and (prev_tokens[-1].text[-1].isalpha() or prev_tokens[-1].text[-1].isdigit()):
-            doc[token.i].is_sent_start = False
-
-        # sentences cannot start with .
-        if token.text == ".":
-            doc[token.i].is_sent_start = False
-
-        # attempt to handle double and quadruple new lines around
-        # section headers by making them their own sentences
-        # for example: '\n\n2 Long Short-Term Memory Networks\n\n\n\n2.1 Overview\n\n'
-        if prev_tokens[-1] and (prev_tokens[-1].text == "\n\n\n\n" or prev_tokens[-1].text == "\n\n"):
-            doc[token.i].is_sent_start = True
-            doc[token.i-1].is_sent_start = True
-
-        # An ! or ? followed by a capital letter should start a sentence
-        if prev_tokens[-1] and prev_tokens[-1].text in ["!", "?"]:
-            if first_char.isupper():
-                doc[token.i].is_sent_start = True
-
-        # don't split up ?! to start a sentence on the !, or vice versa
-        if prev_tokens[-1] and prev_tokens[-1].text in ["?", "!"]:
-            if token.text in ["!", "?"]:
-                doc[token.i].is_sent_start = False
-
-        # update the saved previous tokens
-        prev_tokens = prev_tokens[1:] + [token]
-
-    # any unmatched parens should not be the start of a sentence
-    for open_paren in parens_stack:
-        doc[open_paren.i].is_sent_start = False
-
-    # any unmatched brackets should not be the start of a sentence
-    for open_bracket in brackets_stack:
-        doc[open_bracket.i].is_sent_start = False
+        if token.text.replace('\n', '') == '':
```

Comment:
Is this true in spacy? Maybe good to add a note to the docstring to indicate that newlines cannot be EOS tokens. Might also be worth checking for windows newlines, like `\r` etc.

Reply:
yeah, 
```
>>> doc = nlp("With a\nnew line")
>>> list(doc)
[With, a, 
, new, line]
>>> [token.text.replace('\n', '') for token in doc]
['With', 'a', '', 'new', 'line']
```
I'll add \r as well. Seems pysbd just drops both of those from its output